ledger/tracker.go

```diff
@@ -402,14 +402,33 @@ func (tr *trackerRegistry) scheduleCommit(blockqRound, maxLookback basics.Round)
 	// ( unless we're creating a catchpoint, in which case we want to flush it right away
 	//   so that all the instances of the catchpoint would contain exactly the same data )
 	flushTime := time.Now()
-	if dcc != nil && !flushTime.After(tr.lastFlushTime.Add(balancesFlushInterval)) && !dcc.catchpointFirstStage && !dcc.catchpointSecondStage && dcc.pendingDeltas < pendingDeltasFlushThreshold {
-		dcc = nil
+
+	// Some tracker want to flush
+	if dcc != nil {
+		// skip this flush if none of these conditions met:
+		flushIntervalPassed := flushTime.After(tr.lastFlushTime.Add(balancesFlushInterval))
+		flushForCatchpoint := dcc.catchpointFirstStage || dcc.catchpointSecondStage
+		flushAccounts := dcc.pendingDeltas >= pendingDeltasFlushThreshold
+		if !flushIntervalPassed && !flushForCatchpoint && !flushAccounts {
+			dcc = nil
+		}
 	}
 	tr.mu.RUnlock()
 
 	if dcc != nil {
+		// if deferredCommits cannot accept this task, skip it.
```

Comment:
Move the block comment closer to the line it is referring to, at first I thought it was about the proceeding if not the fact that we are doing a non-blocking enqueue further down. 

```go
if dcc != nil {
    // increment the waitgroup first, otherwise this goroutine can be interrupted
    // and commitSyncer attempts calling Done() on empty wait group.
    tr.accountsWriting.Add(1)
    // do NOT block if deferredCommits cannot accept this task, skip it. 
    // Note: the next attempt will include these rounds plus some extra rounds.
    // the main reason for slow commits is catchpoint file creation,
    // and blocking the producer thread does not make much sense since it has other work to do.
    select {
    case tr.deferredCommits <- dcc:
    default:
        tr.accountsWriting.Done()
}
```